packages/tracing/src/browser/browsertracing.ts

```diff
@@ -230,14 +230,22 @@ export class BrowserTracing implements Integration {
 }
 
 /**
- * Gets transaction context from a sentry-trace meta.
+ * Gets transaction context data from `sentry-trace` and `tracestate` <meta> tags.
  *
- * @returns Transaction context data from the header or undefined if there's no header or the header is malformed
+ * @returns Transaction context data or undefined neither tag exists or has valid data
  */
-export function getHeaderContext(): Partial<TransactionContext> | undefined {
-  const header = getMetaContent('sentry-trace');
-  if (header) {
-    return extractTraceparentData(header);
+export function extractTraceDataFromMetaTags(): Partial<TransactionContext> | undefined {
+  const sentrytraceValue = getMetaContent('sentry-trace');
+  const tracestateValue = getMetaContent('tracestate');
+
+  const sentrytraceData = sentrytraceValue ? extractSentrytraceData(sentrytraceValue) : undefined;
+  const tracestateData = tracestateValue ? extractTracestateData(tracestateValue) : undefined;
+
+  if (sentrytraceData || tracestateData?.sentry || tracestateData?.thirdparty) {
```

Comment:
`tracestateData?.thirdparty` is useless here, as it will never be reached due to the first `sentrytraceData` check already shorting it with it's truthiness.

Reply:
But `sentrytraceData` isn't guaranteed to be truthy - if the `sentry-trace` `<meta>` tag isn't there, or if the value is somehow malformed, you could have `tracestate` data without having `sentry-trace` data. Similarly, if we have a `tracesatate` value in which there's no sentry portion (or the sentry portion is malformed), then we could have third-party `tracestate` data without having sentry `tracestate` data. So I think we need all three checks, since the existence of each value is independent of the existence of the others.